pandas/_libs/src/ujson/lib/ultrajsonenc.c

```diff
@@ -714,6 +714,12 @@ int Buffer_EscapeStringValidated(JSOBJ obj, JSONObjectEncoder *enc,
     }
 }
 
+#define Buffer_Reserve(__enc, __len) \
+    if ( (size_t) ((__enc)->end - (__enc)->offset) < (size_t) (__len))  \
+    {   \
+      Buffer_Realloc((__enc), (__len));\
+    }   \
+
```

Comment:
does it matter that this is moved from the .h file?

Reply:
No this is intentional and matches the ujson source. It essentially reverts part of #17857